### ISSUE TITLE
out of memory: Terminated by signal  9 (SIGKILL)

### ISSUE THREAD

**Issue by ghost** [vlang/v#4703]
**V version:**
V 0.1.26 50a8373.c112b3a
**OS:** 
manjaro linux amd64

**What did you do?**
```v
fn main() {
	for {
		mut s := []int{}
		for i in 1 .. 10000 {
			s << i
		}
	}
}
```

**What did you expect to see?**
run forever, take up to 100mb ram (golang takes 72mb for equivalent)
 
**What did you see instead?**
more than 2gb ram usage and then it crashes with `Terminated by signal  9 (SIGKILL)`

Edit: I see, memory management is marked as work in progress, so maybe this is expected for now.

**Comment by leahlundqvist** [comment]
I don’t think memory is being freed automatically, try adding `s.free()` after the inner for loop and see if that makes a difference

**Comment by medvednikov** [comment]
Good find, I've been working on something related. Will fix.

**Comment by crthpl** [comment]
This still happens with `-autofree`

**Comment by medvednikov** [comment]
I'm not sure why, `s` array is freed after the numbers were added to it.

**Comment by ArtemkaKun** [comment]
@medvednikov 

Alex, I'm not sure why but if you set a capacity for that array - no leak will exist (I checked it with valgrind also)

```v
fn main() {
	for {
		mut s := []int{}

		for i in 1 .. 10000 {
			s << i
		}

		unsafe {s.free()}
	}
}
```

**Comment by spytheman** [comment]
Setting a large enough capacity will make the inner loop not reallocate at all, and freeing `s` in the outer loop, resets things as they were before.

**Comment by ArtemkaKun** [comment]
> Setting a large enough capacity will make the inner loop not reallocate at all, and freeing `s` in the outer loop, resets things as they were before.

Updated my comment because first one was invalid, seems problem related also to endless loop

**Comment by medvednikov** [comment]
Fixed in 0.3.
